python_modules/dagster/dagster/_config/pythonic_config/resource.py

```diff
@@ -867,16 +870,22 @@ def separate_resource_params(cls: Type[BaseModel], data: Dict[str, Any]) -> Sepa
     are marked as resources (ie, using ResourceDependency) from those which are not.
     """
     keys_by_alias = {field.alias: field for field in cls.__fields__.values()}
-    data_with_annotation: List[Tuple[str, Any, Type]] = [
+    data_with_annotation: List[Tuple[str, Any, Type, List[str]]] = [
```

Comment:
This has gotten complex enough that we should a named tuple with four elements so that they are named things. That wiill make line much more readable